tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/bigquery_tests_base.py

```diff
@@ -212,19 +212,9 @@ def assertTableValuesMatch(self, fq_table_name, fields,
                        f"Response returned these values {response_list}")
             self.assertEqual(len(expected_values), len(response_list), message)
 
-            # assert matches for lists of tuple data.  e.g. list of returned fields
-            for value_tuple in expected_values:
-                for result in response_list:
-                    if value_tuple[0] == result[0]:
-                        # have to get the response in this format.  else,
-                        # a comparison of a tuple and a Row object fails
-                        result_tuple = result[:]
-                        message = (
-                            f"Assertion for table {fq_table_name} failed.\n"
-                            f"The result tuple {result_tuple} doesn't match the "
-                            f"expected value tuple {value_tuple}\n{response_list}\n"
-                        )
-                        self.assertEqual(value_tuple, result_tuple, message)
+            # assert matches for lists of tuple data regardless of order.  e.g. list of returned fields
+            result_tuples = [result[:] for result in response_list]
+            self.assertCountEqual(expected_values, result_tuples)
```

Comment:
It's actually a misnomer: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

It checks that the elements are identical akin to `Counter` which uses the dictionary as the underlying.

If you look at the failures, it is actually highlighting legitimate errors in the test data which were not caught earlier.

Reply:
Thank you for the update.  This worked previously because the fields were returned in the order defined by the `fields` list.  The first field was always expected to be an identifier.  Which means the code looped until it found the right row to compare.  Since `assertCountEqual` is comparing dictionaries, it should be able to find matches for the existing tests.